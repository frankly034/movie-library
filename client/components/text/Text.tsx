"use client";

import styled from "styled-components";

interface TextProps {
  readonly textColor?: string;
  readonly size?: string;
}

const Text = styled.p<TextProps>`
  color: ${(props) => props.textColor || "#f9f9f9"};
  font-size: ${(props) => props.size || "14px"};
`;

export default Text;
