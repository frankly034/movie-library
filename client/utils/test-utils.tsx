import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";

import { ThemeProvider } from "../styles";

const Providers = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";
export { customRender as render };
