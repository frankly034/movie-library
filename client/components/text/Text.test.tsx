import { render } from "../../utils/test-utils";

import Text from ".";

describe("Component - Text", () => {
  it("should renders text component", () => {
    const { container } = render(<Text>{"Gift a friend this holiday"}</Text>);
    expect(container).toMatchSnapshot();
  });

  it("should render text", () => {
    const { getByText } = render(<Text>{"Hello world"}</Text>);
    expect(getByText("Hello world")).toBeTruthy;
  });
});
