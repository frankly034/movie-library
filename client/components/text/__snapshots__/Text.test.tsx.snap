// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Title should renders text component 1`] = `
<div>
  <p
    class="sc-bcPKhP gGfpGg"
  >
    Gift a friend this holiday
  </p>
</div>
`;
